### ISSUE TITLE
Removing integration having offer as endpoint results in removal of integrations of all consumers of the offer.

### ISSUE THREAD

**Issue by hemanthnakkina** [juju/terraform-provider-juju#308]
### Description

I have 2 models. One model provides an offer and more than one application from second model consumes the same offer.

If one of the application in second model removes the integration with the offer, the integration is removed from all the applications consumed by the offer.

### Urgency

Blocker for our release

### Terraform Juju Provider version

0.8.0

### Terraform version

v1.5.7

### Terraform Configuration(s)

```terraform
Plans: https://pastebin.ubuntu.com/p/bqHp7z38vh/
```

### Reproduce / Test

Environment: Juju 3.1/stable + microk8s 1.26-strict/stable

```shell
1. On plan1,
terraform init && terraform apply -auto-approve

Look for juju status -m ingress and wait for traefik application is in active state.

2. On plan2,
terraform init && terraform apply -auto-approve

Look for juju status -m sunbeam and wait for all applications are in active state.

3. Remove integration on one of application
# Disable integration with traefik offer on keystone1
echo '{"enable-multiple-traefik-consumers": false}' >> plan2/terraform.tfvars.json
terraform apply -auto-approve.

Look for juju status -m sunbeam. Both keystone, keystone1 applications are in blocked or error state.

Results after running above steps: https://pastebin.ubuntu.com/p/Q3PFmY59nV/

Expected only keystone1 application to be in blocked state and should not trigger anything on keystone application.
```

**Comment by hmlanigan** [comment]
@hemanthnakkina please retry with 0.9.0

**Comment by hmlanigan** [comment]
This bug is not a regression in 0.9.0, will be fixed in 0.10.0
